elftools/dwarf/compileunit.py

```diff
@@ -53,8 +54,13 @@ def __init__(self, header, dwarfinfo, structs, cu_offset, cu_die_offset):
         # requested.
         self._abbrev_table = None
 
-        # A list of DIEs belonging to this CU. Lazily parsed.
+        # A list of DIEs belonging to this CU.
+        # This list is lazily constructed as DIEs are iterated over.
         self._dielist = []
+        # A list of corresponding DIE offsets.
```

Comment:
Can you explain a bit more what each element of `_diemap` is (e.g. its type) and semantics? Is each element an offset from which the corresponding DIE was parsed in the file? So it's the same as DIE.offset? But I guess in a separate list to allow binary searches?

Sicne diemap is a key data structure for the whole approach, it's worth documenting it well. With a good mental model of what diemap is doing, reading the rest of the code becomes easier. "... Show me your tables ..." etc.

Reply:
Sure:

Each element in `_diemap` is an integer, corresponding to a DIE's offset within the file. You're exactly right re: binary searches: the only reason it isn't a `dict` or other type (more closely paring it with `_dielist` is so that `bisect_left` can be used on it.

I'll update the docs to (hopefully) make that more clear.